hudi-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataPayload.java

```diff
@@ -402,6 +422,31 @@ public HoodieMetadataPayload preCombine(HoodieMetadataPayload previousRecord) {
         // 2. A key moved to a different file due to clustering
 
         // No need to merge with previous record index, always pick the latest payload.
+        return this;
+      case METADATA_TYPE_SECONDARY_INDEX:
+        // TODO: This block and checks are just for validation and to detecte all callers.
```

Comment:
This is here mainly for asserting and will be removed later. Ideally all merging logic of secondary (or non-unique-key) index records need to be handled at upper layers directly (when the scanner is running the scan)